kernel/kernel-api/src/main/java/io/delta/kernel/Transaction.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel;
+
+import java.util.List;
+import java.util.Map;
+
+import io.delta.kernel.annotation.Evolving;
+import io.delta.kernel.client.TableClient;
+import io.delta.kernel.data.FilteredColumnarBatch;
+import io.delta.kernel.data.Row;
+import io.delta.kernel.expressions.Literal;
+import io.delta.kernel.types.StructType;
+import io.delta.kernel.utils.*;
+
+import static io.delta.kernel.internal.data.TransactionStateRow.getPartitionColumnsList;
+import static io.delta.kernel.internal.util.PartitionUtils.validatePartitionValues;
+
+/**
+ * Represents a transaction to mutate a Delta table.
+ *
+ * @since 3.2.0
+ */
+@Evolving
+public interface Transaction {
+    /**
+     * Get the schema of the data that is being written to the table.
+     */
+    StructType getSchema(TableClient tableClient);
+
+    /**
+     * Get the list of logical names of the partition columns. This helps the connector to do
+     * physical partitioning of the data before asking the Kernel to stage the data per partition.
+     */
+    List<String> getPartitionColumns(TableClient tableClient);
+
+    /**
+     * Get the state of the transaction. The state helps Kernel do the transformations to logical
```

Comment:
You state how kernel uses this state but you don't really give any information or idea about what the state actually contains. I think giving a simple example or overview would be helpful for the reader to give them some context.

Reply:
I don't think we want the connector to know what is inside. It is a row object that the connector needs to pass it back to Kernel whenever it is needed. Just like the [one](https://github.com/delta-io/delta/blob/master/kernel/kernel-api/src/main/java/io/delta/kernel/Scan.java#L101) we have in Scan side.